File: Java/Array/program8.java
//Accept N numbers from user and display the element having more frequency of occurances

import java.util.*;

class ArrayX
{
    public int Arr[];

    ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    public void Accept()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the elements of array");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public int Frequency(int iValue)
    {
        int iCnt = 0;
        int iFrequency = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == iValue)
            {
                iFrequency++;
            }
        }
        return iFrequency;
    }
}

class program8
{
    public static void main(String arg[])
    {
        int iLength = 0;
        int iNo = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the size of array");
        iLength = sobj.nextInt();

        ArrayX obj = new ArrayX(iLength);

        obj.Accept();

        System.out.println("Enter the element to calculate the frequency");
        iNo = sobj.nextInt();

        iRet = obj.Frequency(iNo);
        System.out.println("Frequency of the element "+ iNo +" is: "+iRet);
    }
}

File: C++/Array/program8.cpp
//Accept N numbers from user and display the element having more frequency of occurances

#include<iostream>
using namespace std;

class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int iLength)
        {
            iSize = iLength;
            Arr = new int[iSize];
        }

        ~ArrayX()
        {
            delete [] Arr;
        }

        void Accept()
        {
            cout<<"Enter the elements : "<<endl;
            for(int iCnt = 0; iCnt < iSize; iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }

        int Frequency(int iValue)
        {
            int iFrequency = 0;

            for(int iCnt = 0; iCnt < iSize; iCnt++)
            {
                if(Arr[iCnt] == iValue)
                {
                    iFrequency++;
                }
            }
            return iFrequency;
        }
};

int main()
{
    int iLength = 0;
    int iNo = 0;
    int iRet = 0;

    cout<<"Enter the size of array"<<endl;
    cin>>iLength;

    ArrayX obj1(iLength);

    obj1.Accept();

    printf("Enter the element to calculate the frequency\n");
    scanf("%d",&iNo);

    iRet = obj1.Frequency(iNo);
    cout<<"Frequency is : "<<iRet;

    return 0;
}